scripts/test/lld.py

```diff
@@ -25,36 +25,53 @@
 def test_wasm_link_metadata():
   print '\n[ checking wasm-link-metadata testcases... ]\n'
 
+  extension_arg_map = {
+    '.json': [],
+    '.jscall.json': ['--emscripten-reserved-function-pointers=3'],
+  }
+
   for obj_path in files_with_pattern(options.binaryen_test, 'lld', '*.o'):
     print '..', obj_path
-    expected_file = obj_path.replace('.o', '.json')
+    for ext, ext_args in extension_arg_map.items():
+      expected_file = obj_path.replace('.o', ext)
+      if ext != '.json' and not os.path.exists(expected_file):
+        continue
 
-    cmd = WASM_LINK_METADATA + [obj_path]
-    actual = run_command(cmd)
+      cmd = WASM_LINK_METADATA + [obj_path] + ext_args
+      actual = run_command(cmd)
 
-    if not os.path.exists(expected_file):
-      print actual
-      fail_with_error('output ' + expected_file + ' does not exist')
-    expected = open(expected_file, 'rb').read()
-    if actual != expected:
-      fail(actual, expected)
+      if not os.path.exists(expected_file):
+        print actual
+        fail_with_error('output ' + expected_file + ' does not exist')
+      expected = open(expected_file, 'rb').read()
+      if actual != expected:
+        fail(actual, expected)
 
 
 def test_wasm_emscripten_finalize():
   print '\n[ checking wasm-emscripten-finalize testcases... ]\n'
 
+  extension_arg_map = {
```

Comment:
Might be good to someday make this a generalized mechanism now that it's used more than just for s2wasm. Not today though?

Reply:
Yeah I agree, we can even unify check.py with auto_update_tests.py because they basically share many processes, but maybe not today though. 😞